.github/workflows/config.yml

```diff
@@ -43,6 +43,13 @@ jobs:
       - name: Install fftw3
         run: |
           sudo apt-get install libfftw3-dev
+          
+      - name: Poetry cache
+        uses: actions/cache@v2
+        with:
+          path: |
+              ~/.cache/pypoetry
+          key: ${{ runner.os }}-${{ hashFiles('./poetry.lock') }}
```

Comment:
Nice.  Hashing the lock file is pretty clever way to decide whether to reinstall the dependencies.